tests/test_hf_api.py

```diff
@@ -223,6 +224,15 @@ def test_create_update_and_delete_repo(self):
         assert res["private"]
         res = self._api.update_repo_visibility(repo_id=repo_id, private=False)
         assert not res["private"]
+
+        # Test gated status update (new functionality)
+        res = self._api.update_repo_settings(repo_id=repo_id, gated="auto")
+        assert res["gated"] == "auto"
```

Comment:
Instead of testing the return value (which is best to remove), you can do 


```py
res = self._api.model_info(repo_id, expand="gated")
```

This will retrieve the information from the server, hence confirming it has been correctly taken into account.